static/src/js/components/Map/Map.jsx

```diff
@@ -286,6 +318,74 @@ const Map = ({
   const mapRef = useRef()
   const mapElRef = useRef()
 
+  useEffect(() => {
+    // Set initial base layer visibility
+    worldImageryLayer.setVisible(base.worldImagery)
+    correctedReflectanceLayer.setVisible(base.trueColor)
+    landWaterMapLayer.setVisible(base.landWaterMap)
+
+    // Set initial overlay layer visibility
+    bordersRoadsLayer.setVisible(overlays.referenceFeatures)
+    coastlinesLayer.setVisible(overlays.coastlines)
+    placeLabelsLayer.setVisible(overlays.referenceLabels)
+  }, [])
+
+  const handleLayerChange = ({ id, checked }) => {
```

Comment:
```suggestion
  const handleLayerChange = ({ id, checked }) => {
    const newBase = {
      worldImagery: false,
      trueColor: false,
      landWaterMap: false
    }
    const newOverlays = { ...overlays }

    // Handle base layers
    if (baseLayerIds.includes(id)) {
      // Set the selected base layer to true
      newBase[id] = checked

      // Update layer visibility
      Object.keys(baseLayers).forEach((layerId) => {
        baseLayers[layerId].setVisible(newBase[id])
      })
    } else {
      // Set overlay layer visibility
      newOverlays[id] = checked
      overlayLayers[id].setVisible(checked)
    }

    // Update redux with the new values
    onChangeMap({
      base: newBase,
      overlays: newOverlays
    })
  }

```

Reply:
`baseLayerIds` would be a new constant added to your mapLayers constant 
```
const baseLayerIds = [
  mapLayers.worldImagery,
  mapLayers.correctedReflectance,
  mapLayers.landWaterMap
]
```

baseLayers and overlayLayers are objects holding the actual layers that make it easy to get them using the ids. They would be defined right after the layers themselves (somewhere like line 187)
```
const baseLayers = {
  [mapLayers.worldImagery]: worldImageryLayer,
  [mapLayers.correctedReflectance]: correctedReflectanceLayer,
  [mapLayers.landWaterMap]: landWaterMapLayer
}

const overlayLayers = {
  [mapLayers.bordersRoads]: bordersRoadsLayer,
  [mapLayers.coastlines]: coastlinesLayer,
  [mapLayers.placeLabels]: placeLabelsLayer
}
```